registry/base_registry.go

```diff
@@ -154,6 +158,32 @@ func (r *BaseRegistry) Register(conf common.URL) error {
 	return nil
 }
 
+// UnRegister
+func (r *BaseRegistry) UnRegister(conf common.URL) error {
+	var (
+		ok  bool
+		err error
+	)
+	r.cltLock.Lock()
+	_, ok = r.services[conf.Key()]
+	r.cltLock.Unlock()
+
+	if !ok {
+		return perrors.Errorf("Path{%s} has not registered", conf.Key())
+	}
+
+	err = r.unregister(conf)
+	if err != nil {
+		return perrors.WithMessagef(err, "register(conf:%+v)", conf)
+	}
+
+	r.cltLock.Lock()
```

Comment:
There is still a time gap between unregister and delete(map)..
I think we should think about:
```go
lock()
service := services[key]
delete(services, key)
unlock
err = service.unregister()
if err != nil 
   lock()
   services[key] = service
   unlock()
```
We should be more careful here.